model/project_ref.go

```diff
@@ -1895,16 +1885,43 @@ func (projectRef *ProjectRef) CanEnableCommitQueue() (bool, error) {
 }
 
 // Upsert updates the project ref in the db if an entry already exists,
-// overwriting the existing ref. If no project ref exists, one is created
-func (projectRef *ProjectRef) Upsert() error {
-	_, err := db.Upsert(
-		ProjectRefCollection,
-		bson.M{
-			ProjectRefIdKey: projectRef.Id,
-		}, projectRef)
+// overwriting the existing ref. If no project ref exists, a new one is created.
+func (p *ProjectRef) Upsert() error {
+	if p.Private == nil {
+		// Projects are private by default unless they've been specially made
+		// public.
+		p.Private = utility.TruePtr()
+	}
+	_, err := db.Upsert(ProjectRefCollection, bson.M{ProjectRefIdKey: p.Id}, p)
```

Comment:
The thing I was thinking would make this safer is using Update instead of Upsert, specifically listing all of the keys that need to be updated (since not everything is implemented in the legacy page). That said if it seems like overkill given the pending deprecation, I'm also fine with that.

Reply:
I think update vs. upsert doesn't matter much in this case (one just happens to insert if it doesn't exist already, the update behavior is otherwise identical between them). It also seems like the upsert won't be going away entirely just by deprecating the legacy UI because it's used in other places like attaching/detaching repos and the REST routes to put/patch a project ref.

I did initially try listing every single key out but it seems like 1. it would be a burden to remember to update the upsert every time we added fields and 2. it would be a headache to keep the data model consistent with the DB model without having extra logic to `$set` and `$unset` as appropriate for zeroing/setting the omitempty fields (if you just `$set` everything, the `omitempty` fields will be `null` instead of nonexistent, which can make some find queries messier since they'll have to know this). Using upsert directly respects the `omitempty` behavior without needing to explicitly `$set`/`$unset` appropriately. I couldn't think of a good reason we would intentionally set `Private` explicitly to false (except via manual DB update), so I thought this was a good enough kind of fix.